src/storage/src/source/postgres/replication.rs

```diff
@@ -390,6 +395,16 @@ async fn raw_stream<'a>(
     >,
     TransientError,
 > {
+    // Skip the timeline ID check for sources without a known timeline ID
+    // (sources created before the timeline ID was added to the source details)
+    if let Some(expected_timeline_id) = timeline_id {
```

Comment:
I think this check wants to look more like where we ensure the replication slot––I have some changes in flight for that in #23746 as inspiration (still fighting with the change)

Reply:
I'm not sure I'm following -- did you mean the check to ensure the replication slot or the publication?

The `ensure_replication_slot` check looks like it returns a TransientError which I don't think makes sense in this context. In your PR it looks like the equivalent of the `ensure_publication_exists` check would involve moving this timeline check to `render` and to do:
```
            if let Err(err) = super::ensure_replication_timeline_id(&replication_client, &connection.publication_details.timeline_id).await? {
                // If the timeline changes there is nothing else to do. These errors
                // are not retractable.
                for &oid in table_info.keys() {
                    let update = ((oid, Err(err.clone())), MzOffset::from(u64::MAX), 1);
                    data_output.give(&data_cap_set[0], update).await;
                }
                return Ok(());
            }
```
but this is effectively the same as what it's doing now since the DefiniteError returned from the `raw_stream` wrapped result is already being handled the same way:
https://github.com/MaterializeInc/materialize/blob/bbe1b090e55fd53f3ebb4d75aa3e8953021e57dd/src/storage/src/source/postgres/replication.rs#L258-L270

Reply:
Sorry I meant publication! And you're totally right––wondering now why that check exists outside of the function? Seems like it would be cleaner (and not have caused the confusion in my comment) if it were inside.